components/org.wso2.carbon.security.caas/src/main/java/org/wso2/carbon/security/caas/user/core/claim/InMemoryClaimManager.java

```diff
@@ -16,23 +16,126 @@
+import org.wso2.carbon.security.caas.user.core.user.UserManager;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 /**
  * InMemory Claim Manager.
  */
 public class InMemoryClaimManager implements ClaimManager {
 
+    private static final Logger log = LoggerFactory.getLogger(InMemoryClaimManager.class);
+
+    /**
+     * IdentityStoreConnector to MetaClaimMappings list.
+     */
+    private Map<String, List<MetaClaimMapping>> metaClaimMappingMap;
+
+    // TODO: Initialize userManager
+    private UserManager userManager;
+
+    // TODO: Initialize metaClaimStore
+    private MetaClaimStore metaClaimStore;
+
     @Override
     public List<Claim> getClaims(User user) throws ClaimManagerException {
-        return null;
+        List<Claim> claims = new ArrayList<>();
+        Domain domain = user.getDomain();
+
+        Map<String, List<MetaClaimMapping>> claimMappings = domain.getClaimMappings();
+
+        Map<String, IdentityStoreConnector> identityStoreConnectors = domain.getIdentityStoreConnectorMap();
+
+
+        for (Map.Entry<String, IdentityStoreConnector> identityStoreConnectorEntry :
+                identityStoreConnectors.entrySet()) {
+
+            String connectorId = identityStoreConnectorEntry.getKey();
+            List<MetaClaimMapping> metaClaimMappings = claimMappings.get(connectorId);
+
+            // Create <AttributeName, MetaClaim> map
+            Map<String, MetaClaim> attributeMapping = metaClaimMappings.stream()
+                    .collect(Collectors.toMap(MetaClaimMapping::getAttributeName, MetaClaimMapping::getMetaClaim));
+
+            IdentityStoreConnector identityStoreConnector = identityStoreConnectorEntry.getValue();
+
+            try {
+                String connectorUserId = userManager.getConnectorUserId(user.getUserId(), connectorId);
+
+                List<Attribute> attributeValues = identityStoreConnector.getUserAttributeValues(connectorUserId,
+                        new ArrayList<>(attributeMapping.keySet()));
+                claims.addAll(buildClaims(attributeValues, attributeMapping));
+            } catch (IdentityStoreException | UserManagerException e) {
+                throw new ClaimManagerException("Error retrieving claims for user : " + user.getUserId(), e);
+            }
+        }
+
+        return claims;
     }
 
     @Override
     public List<Claim> getClaims(User user, List<String> claimURIs) throws ClaimManagerException {
-        return null;
+        List<Claim> claims = new ArrayList<>();
+        Domain domain = user.getDomain();
+
+        Map<String, List<MetaClaimMapping>> claimMappings = domain.getClaimMappings();
+
+        Map<String, IdentityStoreConnector> identityStoreConnectors = domain.getIdentityStoreConnectorMap();
+
+
+        for (Map.Entry<String, IdentityStoreConnector> identityStoreConnectorEntry :
```

Comment:
This can be merged to a single Java 8 stream statement. Extract the try catch block into a single method. IMO it will make the code more clean